engine/src/test/java/net/jqwik/api/EMailsTests.java

```diff
@@ -0,0 +1,208 @@
+		firstPart = firstPart.replace( "_", "");
+		firstPart = firstPart.replace( "`", "");
+		firstPart = firstPart.replace( "{", "");
+		firstPart = firstPart.replace( "|", "");
+		firstPart = firstPart.replace( "}", "");
+		firstPart = firstPart.replace( "~", "");
+		firstPart = firstPart.replace( ".", "");
+		Assertions.assertThat(firstPart).isEqualTo("");
+	}
+
+	@Property
+	void validUseOfDotBeforeAt(@ForAll("emails") String email){
+		String firstPart = getFirstPartOfEmail(email);
+		if(isQuoted(firstPart)){
+			return;
+		}
+		Assertions.assertThat(firstPart).doesNotContain("..");
+		Assertions.assertThat(firstPart.charAt(0)).isNotEqualTo('.');
+		Assertions.assertThat(firstPart.charAt(firstPart.length() - 1)).isNotEqualTo('.');
+	}
+
+	@Property
+	void validSignsAfterAt(@ForAll("emails") String email){
+		String secondPart = getSecondPartOfEmail(email);
+		if(isIPAddress(secondPart)){
+			return;
+		}
+		for(char c = 'a'; c <= 'z'; c++){
+			secondPart = secondPart.replace(c + "", "");
+		}
+		for(char c = 'A'; c <= 'Z'; c++){
+			secondPart = secondPart.replace(c + "", "");
+		}
+		for(char c = '0'; c <= '9'; c++){
+			secondPart = secondPart.replace(c + "", "");
+		}
+		secondPart = secondPart.replace("-", "");
+		secondPart = secondPart.replace(".", "");
+		Assertions.assertThat(secondPart).isEqualTo("");
+	}
+
+	@Property
+	void validUseOfHyphenAndDotAfterAt(@ForAll("emails") String email){
+		String secondPart = getSecondPartOfEmail(email);
+		if(isIPAddress(secondPart)){
+			return;
+		}
+		Assertions.assertThat(secondPart.charAt(0)).isNotEqualTo('-');
+		Assertions.assertThat(secondPart.charAt(secondPart.length() - 1)).isNotEqualTo('-');
+		Assertions.assertThat(secondPart).doesNotContain("..");
+		Assertions.assertThat(secondPart.charAt(0)).isNotEqualTo('.');
+		Assertions.assertThat(secondPart.charAt(secondPart.length() - 1)).isNotEqualTo('.');
+		if(secondPart.length() >= 2){
+			Assertions.assertThat(secondPart.charAt(secondPart.length() - 2)).isNotEqualTo('.');
+		}
+	}
+
+	@Property
+	void validDomainPartLengthAfterAt(@ForAll("emails") String email){
+		String secondPart = getSecondPartOfEmail(email);
+		if(isIPAddress(secondPart)){
+			return;
+		}
+		while (true) {
+			if (secondPart.contains(".")) {
+				int startVal = 0;
+				int checkVal = secondPart.indexOf('.');
+				Assertions.assertThat(checkVal - startVal).isLessThanOrEqualTo(63);
+				secondPart = secondPart.substring(checkVal + 1);
+			} else {
+				Assertions.assertThat(secondPart.length()).isLessThanOrEqualTo(63);
+				return;
+			}
+		}
+	}
+
+	@Property
+	void validIPAddressAfterAt(@ForAll("emails") String email){
+		String secondPart = getSecondPartOfEmail(email);
+		if(!isIPAddress(secondPart)){
+			return;
+		}
+		secondPart = secondPart.substring(1, secondPart.length() - 1);
+		Assertions.assertThat(InetAddresses.isInetAddress(secondPart)).isTrue();
+	}
+
+	@Property
+	void checkFirstPartStatistic(@ForAll("emails") String email){
```

Comment:
I wonder if the name of the method can be improved. what about `quotedAndUnquotedUsernamesAreGenerated`